File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        mp={}
        l,r=0,0
        n=len(nums)
        ans=0
        while r<n:
            mp[nums[r]]=1+mp.get(nums[r],0)
            while mp[nums[r]]>k:
                mp[nums[l]]-=1
                if mp[nums[l]]==0:
                    del mp[nums[l]]
                l+=1
            ans=max(ans,r-l+1)
            r+=1
        return ans

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        int i=0;
        int j=0;
        int len=0;
        unordered_map<int,int> mp;
        while(j<nums.size()){
            mp[nums[j]]++;
            while(mp[nums[j]]>k){
                if(mp[nums[i]]==0) mp.erase(nums[i]);
                mp[nums[i]]--;
                i++;
            }
            len=max(len,j-i+1);
            j++;
        }
        return len;
    }
};
//arr and k 